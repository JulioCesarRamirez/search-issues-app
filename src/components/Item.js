import React from 'react';
import PropTypes from 'prop-types';

export const Item = ({ title, url }) => {
  return (
    <div className="card">
      <a href={url} target="_blank">
        {title}
      </a>
    </div>
  );
};

Item.prototype = {
  issue: PropTypes.string.isRequired,
};
