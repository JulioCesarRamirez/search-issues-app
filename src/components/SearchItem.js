import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const SearchItem = ({ setItemToSearch }) => {
  const [inputValue, setInputValue] = useState('');
  const handleInputChange = ({ target }) => {
    setInputValue(target.value);
  };
  const handleSubmit = (e) => {
    e.preventDefault();
    if (inputValue.trim().length > 2) {
      setItemToSearch((items) => [inputValue, ...items]);
      setInputValue('');
    }
  };
  return (
    <form onSubmit={handleSubmit}>
      <input type="text" value={inputValue} onChange={handleInputChange}/>
    </form>
  );
};

SearchItem.propTypes = {
  setItemToSearch: PropTypes.func.isRequired,
};
