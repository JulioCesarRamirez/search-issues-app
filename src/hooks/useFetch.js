import { useEffect, useState } from "react";
import { getIssues } from "../helpers/getIssues";

export const useFetch = (issue) => {
  const [state, setState] = useState({
    data: [],
    loading: true
  });

  useEffect(() => {
    getIssues(issue).then((items) => {
      setState({ data: items, loading: false });
    });
  }, [issue]);

  return state;
};
