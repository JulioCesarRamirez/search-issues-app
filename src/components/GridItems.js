import React from 'react';
import PropTypes from "prop-types";

import { useFetch } from '../hooks/useFetch';
import { Item } from './Item';

export const GridItems = ({ item }) => {
  const { data: itemsFromApi } = useFetch(item);
  return (
    <div className="card-grid">
      {itemsFromApi.map((item, i) => (
        <Item key={i} {...item} />
      ))}
    </div>
  );
};

GridItems.protoTypes = {
  item: PropTypes.string.isRequired,
};
