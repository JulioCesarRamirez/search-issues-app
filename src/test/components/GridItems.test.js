import React from 'react';
import '@testing-library/jest-dom';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { shallow, configure } from 'enzyme';
import { useFetch } from '../../hooks/useFetch';
import { GridItems } from '../../components/GridItems';

configure({ adapter: new Adapter() });

jest.mock('../../hooks/useFetch');

describe('Test <GridItems />', () => {
  const issue = 'react';

  it('Should show <GridItems /> correctly', () => {
    useFetch.mockReturnValue({
      data: [],
      loading: true,
    });
    const wrapper = shallow(<GridItems item={issue} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should show items when the issues are loaded from useFetch', () => {
    const issues = [
      {
        url: 'https://localhost/anything/thing',
        title: 'Anything',
      },
    ];
    useFetch.mockReturnValue({
      data: issues,
      loading: false,
    });
    const wrapper = shallow(<GridItems items={issue} />);
    expect(wrapper.find('Item').length).toBe(issues.length);
  });
});
