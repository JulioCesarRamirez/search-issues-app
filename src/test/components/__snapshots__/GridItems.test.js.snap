// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <GridItems /> Should show <GridItems /> correctly 1`] = `ShallowWrapper {}`;
